packages/gatsby/cache-dir/app.js

```diff
@@ -34,23 +34,35 @@ apiRunnerAsync(`onClientEntry`).then(() => {
     .then(res => res.json())
     .then(services => {
       if (services.developstatusserver) {
+        let isRestarting = false
         const parentSocket = io(
           `http://${window.location.hostname}:${services.developstatusserver.port}`
         )
 
-        parentSocket.on(`develop:needs-restart`, msg => {
+        parentSocket.on(`structured-log`, msg => {
           if (
+            !isRestarting &&
+            msg.type === `LOG_ACTION` &&
```

Comment:
NB: Is it worth making these constants up top in case we need to change the business logic?